ocaml-lsp-server/src/inference.ml

```diff
@@ -2,7 +2,9 @@ open Import
 
 let infer_intf_for_impl doc =
   match Document.kind doc with
-  | Intf -> failwith "the provided document is not an implementation."
+  | Intf ->
+    Code_error.raise
+      "expected an implementation document, got an interface instead" []
```

Comment:
Is it possible for the user to trigger this error?

Reply:
The input argument is a `Document.t`, and from what I can see the only way to get that is to get the doc from the store. So a user should not be able to trigger this unless there's a bug.